includes/commerce_pos.transaction.inc

```diff
@@ -706,3 +762,89 @@ function commerce_pos_transaction_pay_submit($form, &$form_state) {
   $form_state['transaction']->doAction('save');
   $form_state['redirect'] = 'admin/commerce/pos/' . $form['#redirect_key'] . '/payment';
 }
+
+/**
+ * Validate function for the order search textfield to ensure we have a valid
+ * order number.
+ */
+function _commerce_pos_order_search_validate($form, &$form_state) {
+  if (empty($form_state['values']['order_search']['input'])) {
+    form_set_error('order_search][input', t('Order number cannot be empty.'));
```

Comment:
Remember, Order number and Order id cannot be used interchangeably.

Maybe better to say "Order identifier is required."